project1/application.py

```diff
@@ -29,9 +18,30 @@ def index():
 @app.route("/register", methods=["POST"])
 def show():
     username = request.form.get("username")
+    
+    #to check for the username availability
+    all_users = User.query.all()
+    all_usernames = []
+    for each in all_users:
+        all_usernames.append(each.username)
+
+    if username in all_usernames:
+        return render_template("register.html", flag=True)
+    
     password = request.form.get("pwd")
-    print(username + " " + password)
-    s = "Username: " + username + "\n"  + "Password: "
-    for each in password:
-        s += "*"
-    return s
\ No newline at end of file
+    email = request.form.get("email")
+    gender = request.form.get("gender")
+    DOB = request.form.get("DOB")
+    if DOB == '':
+        DOB = None
+    time_stamp = datetime.now()
+    u = User(username=username, password=password, email=email, gender=gender, DOB=DOB, time_stamp=time_stamp)
+    db.session.add(u)
+    db.session.commit()
+    return render_template("success.html")
```

Comment:
how do you indicate that the registration is failed

Reply:
I am rendering the same registration.html page with the message "Username already taken. Please try with a new one."